File: apps/web/src/types/movie.entity.ts
import { ServerStreamDTO } from './movie.dto';

export interface MovieEntity {
  id: string;
  name: string;
  slug: string;
  originalName: string;
  thumbUrl: string;
  posterUrl: string;
  description: string;
  totalEpisodes: number;
  currentEpisode: string;
  quality: string;
  language: string;
  director: string;
  casts: string;
  duration: string;
  year: string;
  categories: Array<{ id: string; name: string; slug: string }>;
  countries: Array<{ id: string; name: string; slug: string }>;
  servers: ServerStreamDTO[];
}

File: apps/tv-mobile/lib/domain/entities/movie_entity.dart
import 'package:equatable/equatable.dart';

class MovieEntity extends Equatable {
  final String id;
  final String name;
  final String slug;
  final String originalName;
  final String thumbUrl;
  final String posterUrl;
  final String quality;
  final String year;
  final String genre;

  const MovieEntity({
    required this.id,
    required this.name,
    required this.slug,
    required this.originalName,
    required this.thumbUrl,
    required this.posterUrl,
    required this.quality,
    required this.year,
    required this.genre,
  });

  @override
  List<Object?> get props => [id, slug, name, thumbUrl, posterUrl, quality, year, genre];
}
